src/critics/obstacles_critic.cpp

```diff
@@ -22,36 +25,39 @@ void ObstaclesCritic::initialize()
 
 void ObstaclesCritic::score(models::CriticFunctionData & data)
 {
-
   if (!enabled_) {
     return;
   }
 
-  bool all_trajectories_collide = true;
-  for (size_t i = 0; i < data.trajectories.shape(0); ++i) {
-    bool trajectory_collide = false;
+  std::atomic<bool> all_trajectories_collide = true;
```

Comment:
All changes here look fine to me, the question I have is mostly for my education.  I see you added TBB to the xtensor compilation, I imagine that allows for use of multithreading when we ask for xtensor operations. 

But here with `std::for_each` and `std::execution::par_unseq` implies parallelism as well but we don't specify number of threads / really anything. These are new to C++17 which ROS only now moved to so I haven't experimented with this before -- so does this 'just work' more or less without too much of a thought? Or is there more that would need to be done to make this execute in parallel in practice?

Final thought on this is that traditionally when I've done multithreading with TBB/OpenMP there needs to be 'sufficient' amount of work to justify the overhead for multi-threading, or else it runs slower. Have you evalulated the performance here that this actually does improve runtime performance?

Reply:
Yeah, it's just work, and mostly implementation defined. Probably some thread poll would be better alternative here, but without benchmarks i cant notice difference (just with printing control cycles)

Reply:
So just rollback multithreding for now